server/queue/queue-jms/src/test/java/org/apache/james/queue/jms/AbstractJMSMailQueueTest.java

```diff
@@ -115,6 +117,26 @@ public void testFIFO() throws Exception {
     }
 
     @Test
+    public void dequeueShouldPreserveSerializableAttributeClass() throws Exception {
+        // should be empty
+        assertEquals(0, queue.getSize());
+
+        Mail mail = createMail();
+        String name = "name";
+        ImmutableList<String> value = ImmutableList.of("a", "e");
+        mail.setAttribute(name, value);
+
+        queue.enQueue(mail);
+
+        MailQueueItem mailQueueItem = queue.deQueue();
+        assertEquals(mailQueueItem.getMail().getAttribute(name), value);
+        mailQueueItem.done(true);
+
+        // should be empty
+        assertEquals(0, queue.getSize());
```

Comment:
and anyway, this part of the test has nothing to do with test name

Reply:
You're picky. The entire test here is a huge mess, that can not be fixed easily.

The test and the fix still add some value. Should we really enforce ths refactoring today? Is that the wisest choice we can make?

Reply:
You actually wrote this test, you can very easily use assertj and have two tests to check what you want. I didn't asked for a class refactoring, isn't it ?

Reply:
Actually, I have only a single meaninful assert here.

The :

```
+        // should be empty
+        assertEquals(0, queue.getSize());
```

Are here related to this class design. If I remove it, I lead to unreadable , and unrelated test failures, what's makes it useless.

And refactoring this is expensive.

Reply:
Maybe you could explain that assertion so that we understand better your tradeoff.

Reply:
The queue is statefull, and each test should let it empty.

This assertions is here to ensure the test written left the queue "clean".

Reply:
The queue is statefull, and each test should let it empty.

This assertions is here to ensure the test written left the queue "clean".

Reply:
The queue is statefull, and each test should let it empty.

This assertions is here to ensure the test written left the queue "clean".

Reply:
I removed the assert for "clean state" as I empty the queue in  @After